### PR TITLE
Refine styles to better match current dartlang.org

### LINKED ISSUE #1372: add a facelift to dartdoc
We should borrow some of the new css look from dartlang.org.

### PR DESCRIPTION
**Pull request by jcollins-g**
Fixes #1372, #1911.

Resynchronize dartdoc styles with current dartlang.org and fix a number of problems.  Including:

* Smaller fixed-height header
* More space between main content and header
* Larger left-side column
* Use of Roboto font
* Use of `:hover` to highlight links
* Miscellaneous alignment tweaks
* Match color scheme more closely with dartlang.org
* Fix scrolling behavior on mobile (allow overscroll)
* Fix size and scrolling behavior of hamburger menu (no longer exceeds viewport)
* Basic drop shadow for hamburger menu
* Fix viewport parameters for mobile (prevents viewport problems from creating stuck pages)
* In honor of [International Box-Sizing Awareness Day](https://css-tricks.com/international-box-sizing-awareness-day/) set border-box style which I find more intuitive.

Screenshots (slightly larger than window to be able to compare header size)

screenshot from 2019-01-31 10-14-28

screenshot from 2019-01-31 10-15-20

screenshot from 2019-01-31 10-23-04

### CODE REVIEW

File: lib/resources/styles.css [hunk 1]
```diff
@@ -16,6 +16,13 @@ body {
   margin: 0;
   padding: 0;
   height: 100%;
+  width: 100%;
+  overflow: hidden;
+  box-sizing: border-box;
+}
+
+*, *:before, *:after {
```
Comment (devoncarew) [inline lib/resources/styles.css]: This seems somewhat drastic to change every element to box sizing. From very brief googling (https://css-tricks.com/inheriting-box-sizing-probably-slightly-better-best-practice/) this does seem like a thing that people do.

File: lib/resources/styles.css [hunk 2]
```diff
@@ -64,44 +71,44 @@ main {
...
+  margin-top: 5px;
   margin-right: 20px;
 }
 /* end for layout */
 
 body {
   -webkit-text-size-adjust: 100%;
   overflow-x: hidden;
-  font-family: "Source Sans Pro", sans-serif;
+  font-family: roboto, sans-serif;
```
Comment (devoncarew) [inline lib/resources/styles.css]: For similarity with below, `Roboto` (note the case)?

Comment (devoncarew) [reply lib/resources/styles.css]: I assume we're already importing Roboto from somewhere?

I see the import of Source Code Pro and the material icons font, but not of Roboto.

Comment (jcollins-g) [reply lib/resources/styles.css]: Roboto is a builtin as discussed offline

File: lib/resources/styles.css (hunk 1 above)
Comment (jcollins-g) [reply lib/resources/styles.css]: Yes.  Doing this seemed to make correcting some alignment problems for mobile a lot easier.

File: lib/resources/styles.css (hunk 2 above)
Comment (jcollins-g) [reply lib/resources/styles.css]: Fixed capitalization.

**Pull request merged**